### ISSUE TITLE
Configurable syslog targets

### ISSUE THREAD

**Issue by stgraber** [lxc/incus#1420]
Not everyone likes using Grafana Loki for logging. A common alternative which works as a protocol with most log management systems is good old syslog.

It would be good to extend our internal logging/event forwarding mechanism to also work over syslog.

I suspect we may want something like this:
 - logging.syslog.NAME.target (address of the remote syslog server)
 - logging.syslog.NAME.facility (syslog facility to use for the forwarded events)
 - logging.syslog.NAME.severity (syslog severity to use for the forwarded events, if not specified, `info` is used unless we have an internal severity already)
 - logging.syslog.NAME.type (logging, lifecycle or network-acl)
 - logging.syslog.NAME.loglevel (minimum log level when forwarding logging events)

(Once this lands, we'll want to standardize LOKI and move it to `logging.loki.NAME.XYZ` instead of its current single-server top-level at `loki.*`)

**Comment by stgraber** [comment]
@presztak 

Following our discussion this morning and the plan in #1876, I'm going to amend the plan here a bit so we can have a more flexible logging mechanism with multiple remote logging target in simultaneous use.

Here are the list of eventual targets and the target-specific options:

 - syslog
   - address (required)
   - facility (default to `daemon`)
 - loki
   - ca_cert
   - address (required)
   - username
   - password
   - instance (default to local server name)
   - labels
   - retry (number of delivery retries, default 3)
 - webhook
   - ca_cert
   - address (required)
   - username
   - password
   - format (default to `incus`)
   - retry (number of delivery retries, default 3)

And here is the list of what can be logged and any specific option they may have:

 - logging
   - level
 - lifecycle
   - types (e.g. `instance`, comma separate, empty means all)
   - projects (comma separate list of projects, empty means all)
 - network-acl


Here is an example of the suggested structure:

```
logging.loki01.target.type: loki
logging.loki01.target.address: https://loki01.int.example.net
logging.loki01.target.username: foo
logging.loki01.target.password: bar
logging.loki01.types: lifecycle,network-acl
logging.loki01.lifecycle.types: instance

logging.syslog01.target.type: syslog
logging.syslog01.target.address: syslog01.int.example.net
logging.syslog01.target.facility: security
logging.syslog01.types: logging
logging.syslog01.logging.level: warning
```

So from a validation standpoint, we'll need to extract all the keys for a given target name (`loki01` or `syslog01` in this case), then get the target type and based on that we can validate the rest of the configuration for the target.


Let me know if that makes sense or if it looks like I missed something!

**Comment by presztak** [comment]
@stgraber ,

It looks good to me. I have one question. In your first message, you specified config keys in the format:
`logging.syslog.NAME.target`
but in your last message, it’s:
`logging.NAME.target`,
and we also have a `type` key.

Which approach should we follow? The first one might be easier to validate, especially when checking fields for a specific target, but the second one seems more flexible.

**Comment by stgraber** [comment]
Yeah, I switched to the more flexible one when I noticed we'd also be adding webhooks to the mix. 

It's a bit more difficult to validate but will avoid a lot of duplication.
